data-upgrade-news/src/main/java/org/exoplatform/news/upgrade/targets/NewsArticlesStatusUpgrade.java

```diff
@@ -0,0 +1,121 @@
+package org.exoplatform.news.upgrade.targets;
+
+import org.apache.commons.lang3.StringUtils;
+import org.exoplatform.commons.persistence.impl.EntityManagerService;
+import org.exoplatform.commons.upgrade.UpgradeProductPlugin;
+import org.exoplatform.container.PortalContainer;
+import org.exoplatform.container.component.RequestLifeCycle;
+import org.exoplatform.container.xml.InitParams;
+import org.exoplatform.news.model.News;
+import org.exoplatform.news.service.NewsService;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+import org.exoplatform.social.core.jpa.storage.entity.MetadataItemEntity;
+import org.exoplatform.social.metadata.MetadataService;
+import org.exoplatform.social.metadata.model.Metadata;
+import org.exoplatform.social.metadata.model.MetadataType;
+
+import javax.persistence.EntityManager;
+import javax.persistence.Query;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class NewsArticlesStatusUpgrade extends UpgradeProductPlugin {
+  private static final Log         LOG           = ExoLogger.getLogger(NewsArticlesStatusUpgrade.class.getName());
+
+  public static final MetadataType METADATA_TYPE = new MetadataType(4, "newsTarget");
+
+  private EntityManagerService     entityManagerService;
+
+  private NewsService              newsService;
+
+  private MetadataService          metadataService;
+
+  private PortalContainer          container;
+
+  private int                      insertedValue = 0;
+
+  public NewsArticlesStatusUpgrade(InitParams initParams,
+                                   EntityManagerService entityManagerService,
+                                   NewsService newsService,
+                                   MetadataService metadataService,
+                                   PortalContainer container) {
+    super(initParams);
+    this.entityManagerService = entityManagerService;
+    this.newsService = newsService;
+    this.metadataService = metadataService;
+    this.container = container;
+  }
+
+  public int getInsertedValue() {
+    return insertedValue;
+  }
+
+  @Override
+  public void processUpgrade(String oldVersion, String newVersion) {
+    long startupTime = System.currentTimeMillis();
+    LOG.info("Start upgrade of property item status of published news");
+
+    boolean transactionStarted = false;
+    PortalContainer container = PortalContainer.getInstance();
+    RequestLifeCycle.begin(container);
+    EntityManager entityManager = this.entityManagerService.getEntityManager();
+    try {
+      if (!entityManager.getTransaction().isActive()) {
+        entityManager.getTransaction().begin();
+        transactionStarted = true;
+      }
+      List<Metadata> metadataList = metadataService.getMetadatas(METADATA_TYPE.getName(), 0);
+
+      List<MetadataItemEntity> items = new ArrayList<>();
+      if (!metadataList.isEmpty()) {
```

Comment:
this condition is not needed IMO since the next loop will not be considered in that case